### ISSUE TITLE
`set_cursor` effect is delayed on Windows

### ISSUE THREAD

**Issue by kuviman** [tomaka/winit#656]
This can be seen using the `cursor` example. It should change cursor on every key press.

But, on Windows, if you press a key, nothing happens until you move a mouse.

I took a brief look into the code, and it looks like cursor is updated only when the window receives `WM_SETCURSOR` message. According to [the documentation](https://docs.microsoft.com/en-us/windows/desktop/menurc/wm-setcursor), this is sent "if the mouse causes the cursor to move within a window and mouse input is not captured".

Actually, it looks like you capture mouse input simply by holding any mouse button. So, in the same `cursor` example, if you press a key while holding a mouse button, cursor is changed only after you release mouse button.

**Comment by francesca64** [comment]
I believe the cursor setting was deferred like that as an easy way to make sure it ran in the event loop thread (if you try to apply it in the window thread, it doesn't work). These days, `execute_in_thread` should make this easy to fix.

Please give this branch a try: https://github.com/francesca64/winit/tree/win-instant-cursor

**Comment by kuviman** [comment]
@francesca64 Tried it, works perfectly!

**Comment by francesca64** [comment]
I've updated the branch accordingly. How about now?

**Comment by kuviman** [comment]
Looks fine now 👍
